packages/plugin/src/rules/naming-convention.ts

```diff
@@ -8,39 +8,78 @@ const formats = {
   UPPER_CASE: /^[A-Z_]*$/g,
 };
 
-function checkNameFormat(value, style, leadingUnderscore, trailingUnderscore) {
+type ValidNaming = 'camelCase' | 'PascalCase' | 'snake_case' | 'UPPER_CASE';
+const acceptedStyles: ValidNaming[] = ['camelCase', 'PascalCase', 'snake_case', 'UPPER_CASE'];
+interface CheckNameFormatParams {
+  value: string;
+  style?: ValidNaming;
+  leadingUnderscore: 'allow' | 'forbid';
+  trailingUnderscore: 'allow' | 'forbid';
+  suffix: string;
```

Comment:
Any chance you can add here a `prefix` as well? :)

Reply:
This is implemented in https://github.com/dotansimha/graphql-eslint/pull/201/commits/8bc38230826356a0efda5946ea83a0b0e9b07439

Reply:
I'm also thinking that there could be `bannedPrefix` and `bannedSuffix` options as well. Maybe that can go in a separate PR?

Reply:
Yeah, we can add that later, thank you.